utils/legacy_pointer/include/legacy_pointer.hpp

```diff
@@ -0,0 +1,259 @@
+/***************************************************************************
+ *
+ *  Copyright (C) 2016 Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Codeplay's ComputeCpp SDK
+ *
+ *  legacy_pointer.h
+ *
+ *  Description:
+ *    Interface for SYCL buffers to behave as a non-deferrenciable pointer
+ *
+ *
+ **************************************************************************/
+
+#include <CL/sycl.hpp>
+#include <iostream>
+
+namespace codeplay {
+namespace legacy {
+
+/**
+ * PointerMapper
+ *  Associates fake pointers with buffers.
+ *
+ */
+class PointerMapper {
+ public:
+  /* pointer information definitions
+   */
+  const unsigned ADDRESS_BITS = sizeof(void *) * 8;
+  const unsigned BUFFER_ID_BITSIZE = 16u;
+  const unsigned MAX_NUMBER_BUFFERS = 2 << BUFFER_ID_BITSIZE;
+  const unsigned MAX_OFFSET = 2 << (ADDRESS_BITS - BUFFER_ID_BITSIZE);
+
+  using base_ptr_t = uintptr_t;
+
+  /* Fake Pointers are constructed using an integer indexing plus
+   * the offset:
+   *
+   * |== MAX_BUFFERS ==|======== MAX_OFFSET ========|
+   * |   Buffer Id     |       Offset in buffer     |
+   * |=================|============================|
+   */
+  struct legacy_pointer_t {
+    /* Type for the pointers
+    */
+    base_ptr_t _contents;
```

Comment:
I normally would agree but in this case, each util may be coming from different projects (externals), and they may have their own indentation/coding scheme. Can we be more lenient for utils?